verification/policy.go

```diff
@@ -47,107 +46,85 @@ func isPresent(val string, values []string) bool {
 // if any rule is violated, returns an error
 func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
 	// Constants
-	wildcardScope := "*"
+	wildcard := "*"
 	supportedPolicyVersions := []string{"1.0"}
 	supportedVerificationPresets := []string{"strict", "permissive", "audit", "skip"}
 	supportedTrustStorePrefixes := []string{"ca", "signingservice"}
 
 	// Validate Version
 	if !isPresent(policyDoc.Version, supportedPolicyVersions) {
-		return errors.New("Version '" + policyDoc.Version + "' is not supported")
+		return fmt.Errorf("version '%s' is not supported", policyDoc.Version)
 	}
 
 	// Validate the policy according to 1.0 rules
 	if len(policyDoc.TrustPolicies) == 0 {
-		return errors.New("Trust Policy document can not have zero statements")
+		return errors.New("trust policy document can not have zero policy statements")
 	}
-	var uniqueStatementNames []string
+	policyStatementNameCount := make(map[string]int)
 	registryScopeCount := make(map[string]int)
 	for _, statement := range policyDoc.TrustPolicies {
 
 		// Verify statement name is valid
 		if statement.Name == "" {
-			return errors.New("Policy statement is missing a name")
-		} else {
-			if !isPresent(statement.Name, uniqueStatementNames) {
-				uniqueStatementNames = append(uniqueStatementNames, statement.Name)
-			}
+			return errors.New("policy statement is missing a name")
 		}
+		policyStatementNameCount[statement.Name]++
 
 		// Verify registry scopes are valid
 		if len(statement.RegistryScopes) == 0 {
-			return errors.New("Policy statement has zero registry scopes")
-		} else {
-			if len(statement.RegistryScopes) > 1 && isPresent(wildcardScope, statement.RegistryScopes) {
-				return errors.New("Wildcard scope can not be shared with other registry scopes")
-			}
-			for _, scope := range statement.RegistryScopes {
-				registryScopeCount[scope]++
-			}
+			return fmt.Errorf("policy statement '%s' has zero registry scopes", statement.Name)
+		}
+		if len(statement.RegistryScopes) > 1 && isPresent(wildcard, statement.RegistryScopes) {
+			return errors.New("wildcard registry scope can not be shared with other registries")
+		}
+		for _, scope := range statement.RegistryScopes {
+			registryScopeCount[scope]++
 		}
 
 		// Verify signature verification preset is valid
 		if !isPresent(statement.SignatureVerification, supportedVerificationPresets) {
-			return errors.New("SignatureVerification '" + statement.SignatureVerification + "' is not supported")
+			return fmt.Errorf("signatureVerification '%s' is not supported", statement.SignatureVerification)
```

Comment:
nit: `'%s'` can be replaced by `%q`, which automatically quotes strings.

Reply:
Addressed